File: Server/MovieService/MovieServiceDomain/Models/Review.cs
namespace MovieServiceDomain.Models
{
    public class Review : IdModel
    {
        public Guid MovieId { get; private set; }
        public Guid ProfileId { get;  set; }
        public int Value { get; private set; } = 0;

        public string Text { get; private set; } = string.Empty;

        public Review() { }

        public Review(Guid movieId, Guid profileId, int value, string text)
        {
            MovieId = movieId;
            ProfileId = profileId;
            Value = value;
            Text = text;
        }
    }
}

File: Client/reactclient/src/types/review.ts
export interface Review {
  id: string;
  movieId: string;
  profileId: string;
  value: number;
  text: string;
}

export interface CreateReviewRequest {
  accountId: string | null;
  movieId: string;
  value: number;
  text: string;
}

export interface UpdateReviewRequest {
  id: string;
  value: number;
  text: string;
}

export interface PaginatedReview {
  items: Review[];
  pageIndex: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}
